diff --git a/tensorstore/internal/image/jpeg_reader.cc b/tensorstore/internal/image/jpeg_reader.cc
index bfbc67e..8c75a5c 100644
--- a/tensorstore/internal/image/jpeg_reader.cc
+++ b/tensorstore/internal/image/jpeg_reader.cc
@@ -221,7 +221,7 @@ absl::Status JpegReader::Context::Initialize(riegeli::Reader* reader) {
     }
 
     // Read the JPEG header so that we can do error checking.
-    [[maybe_unused]] int code = jpeg_read_header(&cinfo_, /*require_image=*/1);
+    [[maybe_unused]] int code = jpeg_read_header(&cinfo_, static_cast<boolean>(1)); 
     // With `require_image=1` and a non-suspending input source,
     // `jpeg_read_header` cannot return except in the case of success.
     assert(code == JPEG_HEADER_OK);
diff --git a/tensorstore/internal/image/jpeg_writer.cc b/tensorstore/internal/image/jpeg_writer.cc
index b0978bf..6453e84 100644
--- a/tensorstore/internal/image/jpeg_writer.cc
+++ b/tensorstore/internal/image/jpeg_writer.cc
@@ -193,8 +193,8 @@ absl::Status JpegWriter::Encode(const ImageInfo& info,
     }
 
     ::jpeg_set_defaults(&state.cinfo_);
-    ::jpeg_set_quality(&state.cinfo_, options_.quality, /*force_baseline=*/1);
-    ::jpeg_start_compress(&state.cinfo_, /*write_all_tables=*/1);
+    ::jpeg_set_quality(&state.cinfo_, options_.quality, static_cast<boolean>(1));
+    ::jpeg_start_compress(&state.cinfo_, static_cast<boolean>(1));
     state.started_ = true;
 
     while (state.cinfo_.next_scanline < state.cinfo_.image_height) {
