diff --git a/third_party/com_github_nlohmann_json/system.BUILD.bazel b/third_party/com_github_nlohmann_json/system.BUILD.bazel
index 38a02dc..231225c 100644
--- a/third_party/com_github_nlohmann_json/system.BUILD.bazel
+++ b/third_party/com_github_nlohmann_json/system.BUILD.bazel
@@ -1,4 +1,6 @@
 cc_library(
     name = "json",
+    hdrs = glob(["include/nlohmann/**", "include/**"]),
+    includes = ["include"],
     visibility = ["//visibility:public"],
-)
+)
\ No newline at end of file
diff --git a/third_party/com_github_pybind_pybind11/system.BUILD.bazel b/third_party/com_github_pybind_pybind11/system.BUILD.bazel
index 53dee03..0571359 100644
--- a/third_party/com_github_pybind_pybind11/system.BUILD.bazel
+++ b/third_party/com_github_pybind_pybind11/system.BUILD.bazel
@@ -1,9 +1,7 @@
-# NOTE: We could have a config option which copies/symlinks the pybind11
-# headers, as happens in python_configure.bzl
 cc_library(
     name = "pybind11",
+    hdrs = glob(["include/pybind11/**/*.h"]),
+    includes = ["include"],
+    deps = ["@local_config_python//:python_headers"],
     visibility = ["//visibility:public"],
-    deps = [
-        "@local_config_python//:python_headers",
-    ],
 )
diff --git a/third_party/com_google_boringssl/system.BUILD.bazel b/third_party/com_google_boringssl/system.BUILD.bazel
index 2dd9c14..737a795 100644
--- a/third_party/com_google_boringssl/system.BUILD.bazel
+++ b/third_party/com_google_boringssl/system.BUILD.bazel
@@ -1,5 +1,7 @@
 cc_library(
     name = "crypto",
+    hdrs = glob(["include/**/*.h"]),
+    includes = ["include"],
     linkopts = [
         "-lcrypto",
     ],
@@ -8,6 +10,8 @@ cc_library(
 
 cc_library(
     name = "ssl",
+    hdrs = glob(["include/**/*.h"]),
+    includes = ["include"],
     linkopts = [
         "-lssl",
     ],
diff --git a/third_party/jpeg/system.BUILD.bazel b/third_party/jpeg/system.BUILD.bazel
index f9abb8a..8fdc91a 100644
--- a/third_party/jpeg/system.BUILD.bazel
+++ b/third_party/jpeg/system.BUILD.bazel
@@ -1,5 +1,7 @@
 cc_library(
     name = "jpeg",
+    hdrs = glob(["include/**/*.h"]),
+    includes = ["include"],
     linkopts = ["-ljpeg"],
     visibility = ["//visibility:public"],
 )
diff --git a/third_party/libtiff/system.BUILD.bazel b/third_party/libtiff/system.BUILD.bazel
index 6e45467..ad7208a 100644
--- a/third_party/libtiff/system.BUILD.bazel
+++ b/third_party/libtiff/system.BUILD.bazel
@@ -1,5 +1,7 @@
 cc_library(
     name = "tiff",
+    hdrs = glob(["include/**/*.h"]),
+    includes = ["include"],
     linkopts = ["-ltiff"],
     visibility = ["//visibility:public"],
 )
diff --git a/third_party/libwebp/system.BUILD.bazel b/third_party/libwebp/system.BUILD.bazel
index c04ff01..3a3abe4 100644
--- a/third_party/libwebp/system.BUILD.bazel
+++ b/third_party/libwebp/system.BUILD.bazel
@@ -1,5 +1,7 @@
 cc_library(
     name = "webp",
+    hdrs = glob(["include/**/*.h"]),
+    includes = ["include"],
     linkopts = [
         "-lwebp",
         "-lwebpdemux",
diff --git a/third_party/net_zlib/system.BUILD.bazel b/third_party/net_zlib/system.BUILD.bazel
index 12d5f25..e441970 100644
--- a/third_party/net_zlib/system.BUILD.bazel
+++ b/third_party/net_zlib/system.BUILD.bazel
@@ -1,5 +1,7 @@
 cc_library(
     name = "zlib",
+    hdrs = glob(["include/**/*.h"]),
+    includes = ["include"],
     linkopts = ["-lz"],
     visibility = ["//visibility:public"],
 )
diff --git a/third_party/org_aomedia_avif/system.BUILD.bazel b/third_party/org_aomedia_avif/system.BUILD.bazel
index 186a632..4893b54 100644
--- a/third_party/org_aomedia_avif/system.BUILD.bazel
+++ b/third_party/org_aomedia_avif/system.BUILD.bazel
@@ -1,5 +1,7 @@
 cc_library(
     name = "avif",
+    hdrs = glob(["include/**/*.h"]),
+    includes = ["include"],
     linkopts = ["-lavif"],
     visibility = ["//visibility:public"],
 )
diff --git a/third_party/org_blosc_cblosc/system.BUILD.bazel b/third_party/org_blosc_cblosc/system.BUILD.bazel
index 719b82e..305abd2 100644
--- a/third_party/org_blosc_cblosc/system.BUILD.bazel
+++ b/third_party/org_blosc_cblosc/system.BUILD.bazel
@@ -1,5 +1,7 @@
 cc_library(
     name = "blosc",
+    hdrs = glob(["include/**/*.h"]),
+    includes = ["include"],
     linkopts = ["-lblosc"],
     visibility = ["//visibility:public"],
 )
diff --git a/third_party/org_lz4/system.BUILD.bazel b/third_party/org_lz4/system.BUILD.bazel
index 59c86b1..68d96f4 100644
--- a/third_party/org_lz4/system.BUILD.bazel
+++ b/third_party/org_lz4/system.BUILD.bazel
@@ -1,5 +1,7 @@
 cc_library(
     name = "lz4",
+    hdrs = glob(["include/**/*.h"]),
+    includes = ["include"],
     linkopts = ["-llz4"],
     visibility = ["//visibility:public"],
 )
diff --git a/third_party/org_sourceware_bzip2/system.BUILD.bazel b/third_party/org_sourceware_bzip2/system.BUILD.bazel
index aa3a52b..0abb518 100644
--- a/third_party/org_sourceware_bzip2/system.BUILD.bazel
+++ b/third_party/org_sourceware_bzip2/system.BUILD.bazel
@@ -1,5 +1,7 @@
 cc_library(
     name = "bz2lib",
+    hdrs = glob(["include/**/*.h"]),
+    includes = ["include"],
     linkopts = ["-lbz2"],
     visibility = ["//visibility:public"],
 )
diff --git a/third_party/png/system.BUILD.bazel b/third_party/png/system.BUILD.bazel
index a3a5adf..767b944 100644
--- a/third_party/png/system.BUILD.bazel
+++ b/third_party/png/system.BUILD.bazel
@@ -1,5 +1,7 @@
 cc_library(
     name = "png",
+    hdrs = glob(["include/**/*.h"]),
+    includes = ["include"],
     linkopts = ["-lpng"],
     visibility = ["//visibility:public"],
 )
diff --git a/third_party/repo.bzl b/third_party/repo.bzl
index daa3fcb..f9963a6 100644
--- a/third_party/repo.bzl
+++ b/third_party/repo.bzl
@@ -56,6 +56,20 @@ def _handle_use_system_lib(ctx):
             name = ctx.attr.name,
             envvar = SYSTEM_LIBS_ENVVAR,
         ))
+
+    ctx.file(
+        "WORKSPACE.bazel",
+        content = "workspace(name = {})\n".format(repr(ctx.name)),
+        executable = False,
+    )
+
+    prefix = ctx.os.environ.get("PREFIX") or ctx.os.environ.get("CONDA_PREFIX")
+    if prefix:
+        ctx.symlink(prefix + "/include", "include")
+        ctx.symlink(prefix + "/lib", "lib")
+        ctx.symlink(prefix + "/lib64", "lib64")
+
+
     ctx.template(
         "BUILD.bazel",
         ctx.attr.system_build_file,
@@ -133,6 +147,8 @@ _third_party_http_archive = repository_rule(
     attrs = _third_party_http_archive_attrs,
     environ = [
         SYSTEM_LIBS_ENVVAR,
+        "PREFIX",
+        "CONDA_PREFIX",
     ],
 )
 
diff --git a/third_party/se_curl/system.BUILD.bazel b/third_party/se_curl/system.BUILD.bazel
index 17d2bae..228a171 100644
--- a/third_party/se_curl/system.BUILD.bazel
+++ b/third_party/se_curl/system.BUILD.bazel
@@ -1,5 +1,7 @@
 cc_library(
     name = "curl",
+    hdrs = glob(["include/**/*.h"]),
+    includes = ["include"],
     defines = ["TENSORSTORE_SYSTEM_CURL"],
     linkopts = ["-lcurl"],
     visibility = ["//visibility:public"],
diff --git a/third_party/tinyxml2/system.BUILD.bazel b/third_party/tinyxml2/system.BUILD.bazel
index f307f77..5d8aeb4 100644
--- a/third_party/tinyxml2/system.BUILD.bazel
+++ b/third_party/tinyxml2/system.BUILD.bazel
@@ -1,5 +1,7 @@
 cc_library(
     name = "tinyxml2",
+    hdrs = glob(["include/**/*.h"]),
+    includes = ["include"],
     linkopts = ["-ltinyxml2"],
     visibility = ["//visibility:public"],
 )
